sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/PruneFileSourcePartitionsSuite.scala

```diff
@@ -116,7 +107,7 @@ class PruneFileSourcePartitionsSuite extends PrunePartitionSuiteBase {
     // Force datasource v2 for parquet
     withSQLConf((SQLConf.USE_V1_SOURCE_LIST.key, "")) {
       withTempPath { dir =>
-        spark.range(10).selectExpr("id", "id % 3 as p")
+        spark.range(10).coalesce(1).selectExpr("id", "id % 3 as p")
```

Comment:
why need to add `coalesce`?

Reply:
Same question here. Looks good otherwise.

Reply:
@viirya @huaxingao this is because `TestHiveSingleton` uses `local[1]` while `SharedSparkSession` uses `local[2]` so it will generate 2 Parquet files. As result, `ParquetScan` will generate 2 input splits which makes the test fail since the provided value for `expectedPartitionCount` is 1.